Examples/src/main/kotlin/com/mapbox/vision/examples/POIActivityKt.kt

```diff
@@ -0,0 +1,153 @@
+import com.mapbox.vision.mobile.core.models.road.RoadDescription
+import com.mapbox.vision.mobile.core.models.world.WorldDescription
+import java.io.IOException
+import java.io.InputStream
+import kotlinx.android.synthetic.main.activity_main.vision_view
+import kotlinx.android.synthetic.main.activity_poi.*
+
+class POIActivityKt : BaseActivity() {
+
+    private var visionReplayManagerWasInit = false
+
+    // Download session from tutorial and push to device
+    private var sessionPath = "${Environment.getExternalStorageDirectory().absolutePath}/session"
+
+    private val visionEventsListener = object : VisionEventsListener {
+
+        private var cameraCalibrated = false
+
+        override fun onAuthorizationStatusUpdated(authorizationStatus: AuthorizationStatus) {}
+
+        override fun onFrameSegmentationUpdated(frameSegmentation: FrameSegmentation) {}
+
+        override fun onFrameDetectionsUpdated(frameDetections: FrameDetections) {}
+
+        override fun onFrameSignClassificationsUpdated(frameSignClassifications: FrameSignClassifications) {}
+
+        override fun onRoadDescriptionUpdated(roadDescription: RoadDescription) {}
+
+        override fun onWorldDescriptionUpdated(worldDescription: WorldDescription) {}
+
+        override fun onVehicleStateUpdated(vehicleState: VehicleState) {
+            if (cameraCalibrated) {
+                poi_view.updatePOIState(vehicleState.geoLocation)
+            }
+        }
+
+        override fun onCameraUpdated(camera: Camera) {
+            if (camera.calibrationProgress == 1.0f) {
+                cameraCalibrated = true
+                poi_view.onCameraCalibrated(camera.frameWidth, camera.frameHeight)
+            }
+        }
+
+        override fun onCountryUpdated(country: Country) {}
+
+        override fun onUpdateCompleted() {}
+    }
+
+    override fun onPermissionsGranted() {
+        startVisionManager()
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_poi)
+        poi_view.poiList = buildPOIList()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        startVisionManager()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        stopVisionManager()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        vision_view.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        vision_view.onPause()
+    }
+
+    private fun startVisionManager() {
+        if (allPermissionsGranted() && !visionReplayManagerWasInit) {
+            VisionReplayManager.create(sessionPath)
+            VisionReplayManager.visionEventsListener = visionEventsListener
+            vision_view.setVisionManager(VisionReplayManager)
+            VisionReplayManager.start()
+            visionReplayManagerWasInit = true
+        }
+    }
+
+    data class POI(
+        val longitude: Double,
+        val latitude: Double,
+        val bitmap: Bitmap?
```

Comment:
BTW is there any point if `Bitmap` actually is null? It will not be shown I guess and it could be confusing.
Perhaps it would be better to throw `RuntimeException` if we could not get `Bitmap` from assets and make it non-nullable here.

Reply:
fixed